src/components/Explore/Map.tsx

```diff
@@ -135,16 +135,22 @@ const ExploreMap = (): JSX.Element => {
                 if (e.features && e.features[0]) {
                     const feature = e.features[0];
                     const stationProperties = feature.properties as StationSummary;
-                    const newSelectedStation =
+                    let newSelectedStation =
                         stationProperties.name === selectedStationRef.current?.name ? null : stationProperties;
+                    if (newSelectedStation) {
+                        const index: number = stationsList
```

Comment:
A suggestion for a micro-enhancement of performance! Instead of chaining `map` and `indexOf`, you can use `findIndex` so the code only iterates over the list once:

`stationsList.findIndex(({ name }) => newSelectedStation.name === name);`

Reply:
Ohh :P, I forgot I can leverage object destructuring here! The reason why I had to do that was because of the object of `StationSummary`. I'll rectify it now.